### PR TITLE
Adds http(s) support to openshift.process and adds openshift.cluster()

### LINKED ISSUE #63: openshift.process should support http(s) urls
openshift.process should support http(s) urls by using the -f flag, similar to how openshift.(verb) works.

Code that needs to be fixed:
https://github.com/openshift/jenkins-client-plugin/blob/master/src/main/resources/com/openshift/jenkins/plugins/OpenShiftDSL.groovy#L570-L604

Working example:
https://github.com/openshift/jenkins-client-plugin/blob/master/src/main/resources/com/openshift/jenkins/plugins/OpenShiftDSL.groovy#L523-L556

### PR DESCRIPTION
**Pull request by coreydaley**
openshift.process can now accept an http/https url to a template
openshift.cluster returns the url of the cluster being used

Fixes #63 & #64

### CODE REVIEW

**Comment by coreydaley** [comment]
@openshift/sig-developer-experience ptal

**Comment by coreydaley** [comment]
[test]

**Comment by coreydaley** [comment]
@gabemontero 
Addressed item 1 in above comment, added some openshift.cluster() usage to examples

**Comment by gabemontero** [comment]
I'll let the test complete, then merge.

Probably ought to wait on 3.7 shipping before cutting a new version of the plugin at this point, but we can discuss that with @bparees either here or in scrum

**Comment by gabemontero** [comment]
[merge]

File: src/main/resources/com/openshift/jenkins/plugins/OpenShiftDSL.groovy [hunk 1]
```diff
@@ -582,21 +584,24 @@ class OpenShiftDSL implements Serializable {
...
+        if ( markup && !httpref ) { // does this look like json or yaml?
+            FilePath f = currentContext.exec.getWorkspaceFilePath().createTextTempFile( "process", ".markup", s, false );
             try {
-                r = new Result( "process" );
                 r.actions.add((OcAction.OcActionResult)script._OcAction( buildCommonArgs("process", ["-f", f.getRemote() ], args, "-o=json" ) ));
                 r.failIf( "process returned an error" );
             } finally {
                 f.delete();
             }
+        } else if (httpref) {
```
Comment (bparees) [inline src/main/resources/com/openshift/jenkins/plugins/OpenShiftDSL.groovy]: if it starts with a { why would we ever treat it as a http url?

Comment (bparees) [reply src/main/resources/com/openshift/jenkins/plugins/OpenShiftDSL.groovy]: ok i guess i see how the logic works.  looks fragile but i guess this isn't making it worse.

**Comment by bparees** [comment]
> Probably ought to wait on 3.7 shipping before cutting a new version of the plugin at this point, but we can discuss that with @bparees either here or in scrum

this seems like a bug people are going to hit, i'd rather fix it before we ship.

**Comment by gabemontero** [comment]
10-4 I'll start the update process

**Comment by gabemontero** [comment]
awaiting v1.0.1 at https://updates.jenkins-ci.org/download/plugins/openshift-client/

File: src/main/resources/com/openshift/jenkins/plugins/OpenShiftDSL.groovy (hunk 1 above)
Comment (coreydaley) [reply src/main/resources/com/openshift/jenkins/plugins/OpenShiftDSL.groovy]: I tried to duplicate what was there, i plan on doing some further refactoring but didn't want to mix that in this pr

**Comment by gabemontero** [comment]
centos image update https://github.com/openshift/jenkins/pull/408

**Comment by gabemontero** [comment]
rhel image update https://buildvm.openshift.eng.bos.redhat.com:8443/job/devex/job/devex%252Fjenkins-plugins/8/console

**Pull request merged**